app/ui-react/syndesis/src/modules/integrations/components/editor/utils.ts

```diff
@@ -42,6 +44,44 @@ type StepKindHrefCallback = (
   s: ISelectConnectionRouteState
 ) => H.LocationDescriptorObject;
 
+/**
+ * Checks whether or not the provided object is a technical preview.
+ * Accepts either an IUIStep, Connection, or a Connector.
+ * Returns a boolean for whether or not the metadata `tech-preview` key
+ * returns a string value of 'true'
+ * @param data
+ */
+export function figureOutTechPreviewFlag<
+  T extends IUIStep | Connection | Connector
+>(data: T): boolean {
+  if ((data as IUIStep).connection && (data as IUIStep).connection!.connector) {
+    return (
+      getMetadataValue<string>(
+        'tech-preview',
+        (data as IUIStep).connection!.connector!.metadata
+      ) === 'true'
+    );
+  } else if ((data as IUIStep).connector) {
```

Comment:
So, you don't probably need this `if` statement, or the next one, at runtime this type info is meaningless 'cause it's gone.

Reply:
If I do this instead, won't it always return `false` then for whenever the object is a `Connection` or is that a really dumb question? 😖 
```
export function figureOutTechPreviewFlag<
  T extends IUIStep | Connection | Connector
>(data: T): boolean {
  if ((data as IUIStep).connection && (data as IUIStep).connection!.connector) {
    return (
      getMetadataValue<string>(
        'tech-preview',
        (data as IUIStep).connection!.connector!.metadata
      ) === 'true'
    );
  } else if (data as Connector) {
    return getMetadataValue<string>('tech-preview', data.metadata) === 'true';
  } else {
    return false;
  }
}
```

Reply:
I guess my question would be, then why would we want to accept `Connection` as a type? Would it make more sense to only accept `IUIStep` and `Connector`?

Reply:
Hmm I suppose in places that affects, like the Dashboard Connections list, I can pass a connector instead of connection, but the last question still applies I think.